plugin/aws/ec2Instance.go

```diff
@@ -238,457 +191,13 @@ func (inst *ec2Instance) checkLatestConsoleOutput(ctx context.Context) (*ec2Inst
-			// cancelled (i.e. when reason == ctx.Err().Error()).
-			cancelCtx, cancelFunc := context.WithTimeout(context.Background(), 3*time.Second)
-			defer cancelFunc()
-
-			// CancelCommand's response doesn't contain any useful information,
-			// so we can discard it.
-			_, err := inst.ssmClient.CancelCommandWithContext(cancelCtx, request)
-			if err != nil {
-				activity.Record(ctx, "Failed to cancel the command: %v", err)
-			} else {
-				activity.Record(ctx, "Successfully cancelled the command")
-			}
-		}
-
-		// According to https://github.com/aws/amazon-ssm-agent/blob/2.3.479.0/agent/agentlogstocloudwatch/cloudwatchlogspublisher/cloudwatchlogsservice.go#L40,
-		// SSM updates the stdout/stderr logs every three seconds,
-		// so we'll sleep a little bit more than that between each
-		// iteration to account for delays
-		sleepDuration := 4 * time.Second
-
-		for {
-			time.Sleep(sleepDuration)
-
-			activity.Record(ctx, "Getting the command status")
-			request := &ssm.GetCommandInvocationInput{
-				CommandId:  awsSDK.String(commandID),
-				InstanceId: awsSDK.String(inst.id),
-			}
-
-			resp, err := inst.ssmClient.GetCommandInvocationWithContext(ctx, request)
-			if err != nil {
-				err := fmt.Errorf("failed to get the command status: %v", err)
-				cancelCommand(err.Error())
-				closeOutputStreamer(err)
-				return
-			}
-
-			status := awsSDK.StringValue(resp.StatusDetails)
-			activity.Record(ctx, "Command status: %v", status)
-
-			switch status {
-			case CommandPending, CommandDelayed:
-				continue
-			case CommandInProgress:
-				if _, err := outputStreamer.sendNextChunks(); err != nil {
-					cancelCommand(err.Error())
-					closeOutputStreamer(err)
-					return
-				}
-			case CommandSuccess, CommandFailed:
-				exitCode = int(awsSDK.Int64Value(resp.ResponseCode))
-
-				// Write the remaining output chunks (if any)
-				activity.Record(ctx, "Command finished. Sending over its remaining output...")
-				for {
-					chunksWereSent, err := outputStreamer.sendNextChunks()
-					if err != nil {
-						activity.Record(ctx, "Failed to send over the remaining output: %v", err)
-						closeOutputStreamer(err)
-						return
-					}
-
-					if chunksWereSent {
-						time.Sleep(sleepDuration)
-					} else {
-						// No new chunks were written. Assume that this means
-						// we've finished writing the output
-						activity.Record(ctx, "Finished sending the output.")
-						closeOutputStreamer(nil)
-						return
-					}
-				}
-			case CommandExecutionTimedOut:
-				closeOutputStreamer(fmt.Errorf("the command timed out during its execution"))
-				return
-			default:
-				closeOutputStreamer(fmt.Errorf("the command failed to start. Its status is %v", status))
-				return
-			}
-		}
-	}()
-
-	execResult.OutputCh = outputCh
-	execResult.ExitCodeCB = func() (int, error) {
-		if exitCodeErr != nil {
-			return 0, exitCodeErr
-		}
-
-		return exitCode, nil
-	}
+	dnsname := meta["PublicDnsName"].(string)
```

Comment:
This may not be configured. Need to try public DNS name, then public IP, then error cleanly.